projects/storefrontlib/src/lib/ui/pages/category-page/category-page.component.ts

```diff
@@ -1,37 +1,40 @@
-import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute, Params } from '@angular/router';
+import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
 import { Observable } from 'rxjs';
-import { CmsService, Page } from '@spartacus/core';
+import { CmsService } from '@spartacus/core';
+import { map } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-category-page',
-  templateUrl: './category-page.component.html',
-  styleUrls: ['./category-page.component.scss']
+  templateUrl: './category-page.component.html'
 })
-export class CategoryPageComponent implements OnInit {
-  categoryCode: string;
-  brandCode: string;
-  query: string;
-  cmsPage$: Observable<Page>;
-
+export class CategoryPageComponent {
   constructor(
     protected activeRoute: ActivatedRoute,
     protected cmsService: CmsService
   ) {}
 
-  ngOnInit() {
-    this.activeRoute.params.forEach((params: Params) => {
-      if (params['categoryCode']) {
-        this.categoryCode = params['categoryCode'];
-      }
-      if (params['brandCode']) {
-        this.brandCode = params['brandCode'];
-      }
-      if (params['query']) {
-        this.query = params['query'];
-      }
-    });
+  get categoryCode(): Observable<string> {
```

Comment:
It's a bagatelle: it would be good to end all those observables with `$` to follow the convention,